src/auxinfo/participant.rs

```diff
@@ -110,6 +121,10 @@ impl ProtocolParticipant for AuxInfoParticipant {
     ) -> Result<ProcessOutcome<Self::Output>> {
         info!("Processing auxinfo message.");
 
+        if *self.status() == Status::TerminatedSuccessfully {
+            return Err(InternalError::ProtocolAlreadyTerminated);
```

Comment:
question: What would cause this error to be thrown? I want to try to consolidate our errors into, like, "the caller messed up" vs "somebody else messed up", but I guess this one could go either way, right?

Reply:
I'm hoping this would just be a "Caller messed up" error. However, I need to figure out what's going on with `BroadcastParticipant` to determine if this is true :-)

Reply:
It could also be caused by a malicious party doing something weird like sending messages more than once, though. Then even if the caller routed everything it received correctly, we might still have this problem. I thought we might try to consolidate this into one of our existing errors but now I think we should keep it as a separate case from either `ProtocolError` or `CallerError` since we can't tell internally which one it is.

No action required.